lucene/directory-provider/src/test/java/org/infinispan/hibernate/search/TwoNodesTest.java

```diff
@@ -60,6 +60,7 @@ public void testSomething() {
       } finally {
          nodeC.close();
       }
+      waitMembersCount(nodea, SimpleEmail.class, 2);
```

Comment:
This method is based on a counter: https://github.com/infinispan/infinispan/blob/fd9410e09233baaab91861b96d1210bd4c6728ad/lucene/directory-provider/src/test/java/org/infinispan/hibernate/search/ClusterTestHelper.java#L96

However I totally agree - it should be based on time.